src/attribute.ts

```diff
@@ -29,25 +29,32 @@ export class Attribute {
   parent?: Attribute
   children: Attribute[]
   slices: Attribute[]
+  items: Attribute[]
 
+  id: string
   name: string
+  definition: AttributeDefinition
+  types: string[]
+
   isArray: boolean
   isSlice: boolean
+  isItem: boolean
   isRequired: boolean
   isPrimitive: boolean
-  definition: AttributeDefinition
-  types: string[]
+
   index?: number
 
-  constructor(definition: AttributeDefinition, index?: number) {
+  constructor(definition: AttributeDefinition) {
     this.children = []
     this.slices = []
+    this.items = []
 
     this.definition = definition
-    this.name = definition.id.split('.').pop()!
-    this.index = index
-    this.isArray = index === undefined && definition.max !== '1'
+    this.id = definition.id.split('.').pop()!
+    this.name = definition.path.split('.').pop()!
+    this.isArray = definition.max === '*' || Number(definition.max) > 1
     this.isSlice = !!definition.sliceName
+    this.isItem = false
```

Comment:
I find the usage of isItem a bit weird. Couldn't you use parent.isArray instead?

Reply:
as explained orally, the array is no longer the parent of its items